markdown/opentelemetry-intro/opentelemetry-intro.md

```diff
@@ -0,0 +1,319 @@
+
+Use a text editor to save the the connection parameters with the **Host**, **Message VPN Name**, **Client Username** and **Password**. You'll need them later.
+
+✔️ You've created the first part of the solution!
+![check1](img/check1.png)
+
+## Set up Jaeger All-In-One
+
+Duration: 0:05:00
+
+Jaeger collects and visualizes the spans created by your microservices.  It has several different components such as an underlying database, a GUI and a span collector, but fortunately it has an all-in-one distribution that makes setup easy.
+
+Negative
+: If you don't have a tool that opens .gz files, you can download one. [7-Zip](https://www.7-zip.org/download.html) is a good free option
+
+### Steps to install
+1. Extract the Jaeger distribution file you downloaded as part of the prerequisites.  (e.g. `jaeger-1.17.1-windows-amd64.tar.gz` on Windows)
+1. Open a command terminal.
+1. Run `jaeger-all-in-one`
+1. Using a web browser, navigate to `http://localhost:16686/`
+1. You should be greeted by a picture of a...chipmunk? Gopher?  Not really sure.
+
+Negative
+: You'll need to keep the terminal window running Jaeger open.  If you close it, you'll lose the traces you've collected.
+
+✔️ You've created the second part of the solution!
+![check2](img/check2.png)
+## Set up Python environment
+
+Duration: 0:12:00
+
+Python is a great language for prototyping applications.  It's easy to get up and running quickly, and since it doesn't compile you just change a line and reload.  This CodeLab uses Python to simulate Salesforce publishing the message and two event consumers using the Solace PubSub+ Python API.  And since there is a native OpenTelemetry API for Python it can produce the OpenTelemetry tracing as well.  
+
+### Steps to install
+1. Open the Python installer that you downloaded in the prerequisites.
+1. You'll be guided through the install process. Be sure to include Python in your PATH variable.
+
+Once it's complete, you should have a program called IDLE (Integrated Development and Learning Environment).
+![IDLE](img/idle.png "IDLE")
+  That's what you'll use to modify and run the Python scripts.
+## Create the "Salesforce" message publisher
+
+Duration: 0:10:00
+
+Positive
+: This CodeLab focuses on OpenTelemetry, so it won't actually implement the Salesforce events.  If that's your passion, Solace has a great [CodeLab about event-enabling Salesforce with Boomi and Solace](https://codelabs.solace.dev/codelabs/boomi-salesforce-codelab/).  Likewise, this CodeLab doesn't get into the nitty gritty of the Python API.  If you want to know more about those capabilities, there will be a CodeLab soon.
+
+1. If you haven't already, unzip the CodeLabs code into a convenient directory
+1. Open a terminal window 
+1. Install the OpenTelemetry APIs with pip:
+
+		pip install -r requirements.txt
+		
+1. Launch IDLE, and open `solace_telemetry_publisher_Salesforce.py`
+1. Update the connection parameters to match your Solace event broker:
```

Comment:
Since we are using environment variables, I wold recommend changing the instructions here to include setting up the env variables in windows 

Windows: https://www.clear.rice.edu/engi128/Resources/usingidle.htm
MacOS: Just pass from terminal